pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpStream.java

```diff
@@ -63,77 +67,87 @@ public MeetupRsvpStream(boolean partitionByKey)
     properties.put("serializer.class", "kafka.serializer.DefaultEncoder");
     properties.put("request.required.acks", "1");
     _producer = StreamDataProvider.getStreamDataProducer(KafkaStarterUtils.KAFKA_PRODUCER_CLASS_NAME, properties);
-  }
-
-  public MeetupRsvpStream(boolean partitionByKey, StreamDataProducer producer, String topicName) {
-    _partitionByKey = partitionByKey;
-    _producer = producer;
-    _topicName = topicName;
+    _source = new Source(createConsumer());
   }
 
   public void run()
       throws Exception {
-    _client = ClientManager.createClient();
-    _keepPublishing = true;
-
-    _client.connectToServer(new Endpoint() {
-      @Override
-      public void onOpen(Session session, EndpointConfig config) {
-        session.addMessageHandler(String.class, getMessageHandler());
-      }
-    }, ClientEndpointConfig.Builder.create().build(), new URI("wss://stream.meetup.com/2/rsvps"));
+    _source.start();
```

Comment:
i felt like this original construct was to demonstrate how to create a listening client against some public API and redirect data into pinot.

If we are not doing such it this can simply be a periodic task that runs `_producer.produce(createMessage)` without the source/consumer wrapper